network_simulator/host.py

```diff
@@ -23,4 +28,91 @@ def __init__(self, address=None, flows={}, flow_packets_received={},
         self.flow_packets_received = flow_packets_received
         self.link = link
 
-    # TODO(team): More event subclasses to come?
\ No newline at end of file
+
+class HostReceivedPacketEvent(Event):
+    """
+    Host receives packets from a flow.
+    """
+
+    def __init__(self, host, packet, packet_previously_received):
+        """
+        Sanity check: make sure destination was this Host and that this isn't
+        a RouterPacket.
+        :ivar host: Host that receives Packet
+        :ivar packet: Packet received
+        :ivar packet_previously_received: True if Packet previously received
+        """
+        Event.__init__(self)
+
+        # Sanity check
+        assert packet.dest_id == host.address and not isinstance(packet,
+                                                                 RouterPacket)
+        self.host = host
+        self.packet = packet
+        self.packet_previously_received = packet_previously_received
+
+    def elem_in_list(elem, my_list):
+        # ind is where this element *would* be inserted
```

Comment:
You need triple-quote (i.e. """) comments explaining the function and its parameters, as the very first line. Format is:
```
"""
<explanation of what function does>
:param elem: <description>
:param my_list: <description>
```